EnvChanger/Classes/EnvChangerController.swift

```diff
@@ -32,45 +32,103 @@ public class EnvChangerController<T>: UIViewController where T: EnvironmentRepre
+        return alert
+    }
+    
+    private var environmentChangedAlert: UIAlertController {
+        guard let savedEnvironment = savedEnvironment else { return UIAlertController() }
+        
+        let alert = UIAlertController(title: "Environment successfully changed.",
+                                      message: "Your active environment is now: \n \(savedEnvironment)",
+                                      preferredStyle: .alert)
+        alert.addAction(UIAlertAction(title: "Great!", style: .default, handler: nil))
+        return alert
+    }
     
     /// Instantiate with the object of environments you would like to access.
     ///
     /// - Parameters:
     ///   - envs: 'T' of type String, CaseIterable object that should preferably have environments inside.
     ///   - buttonConfiguration: Sets the button title or image that will be displayed.
-    ///   - completionHandler: Add any logic you would want to execute after you selected your new environment in the completionHandler.
+    ///   - environmentSelectedHandler: Add any logic you would want to execute after you selected your new environment in the environmentSelectedHandler.
     /// - Note:
     ///   - If no buttonConfiguration in the constructor by default it will set the title to 'EN'.
     ///   - Saves the first environment passed in to avoid getSavedEnvironment() to return an empty string.
     public init(envs: T.Type,
+                window: UIWindow?,
                 buttonConfiguration: EnvButtonConfiguration = .title("ENV"),
-                completionHandler: @escaping (T) -> Void) {
-        super.init(nibName: nil, bundle: nil)
-        self.envs = envs
+                environmentSelectedHandler: @escaping (T) -> Void) {
+        self.appWindow = window
         self.buttonConfiguration = buttonConfiguration
-        self.completionHandler = completionHandler
-        
+        self.envs = envs
+        self.environmentSelectedHandler = environmentSelectedHandler
+        super.init(nibName: nil, bundle: nil)
         setupWindow()
         saveFirstEnvironment()
     }
     
-    override public func loadView() {
-        setupUI()
-    }
-    
     required init?(coder aDecoder: NSCoder) {
         fatalError()
     }
     
-    public func getSavedEnvironment() -> String {
-        return UserDefaults.standard.string(forKey: ACTIVE_ENV_KEY) ?? ""
+    @objc private func panDidFire(panner: UIPanGestureRecognizer) {
+        let offset = panner.translation(in: view)
+        panner.setTranslation(.zero, in: view)
+        var center = button.center
+        center.x += offset.x
+        center.y += offset.y
+        button.center = center
+    }
+    
+    @objc private func environmentButtonTapped() {
+        /// Prevents from attempting to present the same alert when clicking on the ENV change button.
+        guard appWindow?.rootViewController?.presentedViewController as? UIAlertController == nil else { return }
```

Comment:
[optional]

`if appWindow?.rootViewController?.presentedViewController is UIAlertController { return }`
This way you avoid the `== nil`.

Reply:
Done.